Source/Urho3D/CSharp/Swig/strings.i

```diff
@@ -13,19 +13,88 @@
 #include <EASTL/string.h>
 %}
 
-namespace eastl {
+%wrapper %{
+    SWIGEXPORT void SWIGSTDCALL CSharp_Urho3D_String_Set(eastl::string* str, const char* val) { *str = val; }
+    SWIGEXPORT const char* SWIGSTDCALL CSharp_EaStl_String_Get(eastl::string* str) { return str->data(); }
+%}
 
-%naturalvar string;
-%naturalvar string_view;
+%pragma(csharp) modulecode=%{
+    [System.Runtime.InteropServices.DllImport($dllimport, EntryPoint="CSharp_Urho3D_String_Set")]
+    internal static extern void SetString(global::System.IntPtr str,
+        [param: System.Runtime.InteropServices.MarshalAs(global::Urho3DNet.Urho3DPINVOKE.LPStr)]
+        string val);
+
+    [System.Runtime.InteropServices.DllImport($dllimport, EntryPoint="CSharp_Urho3D_String_Get")]
+    [return: System.Runtime.InteropServices.MarshalAs(global::Urho3DNet.Urho3DPINVOKE.LPStr)]
+    internal static extern string GetString(global::System.IntPtr str);
+
+    public static unsafe System.IntPtr strdup(string str)
+    {
+        int len = System.Text.Encoding.UTF8.GetByteCount(str) + 1;
+        System.IntPtr ptr = Urho3DNet.Urho3D.malloc(len);
+        byte* pBytes = (byte*)ptr;
+        fixed (char* pChar = str) {
+            System.Text.Encoding.UTF8.GetBytes(pChar, str.Length, pBytes, len - 1);
+            pBytes[len - 1] = 0;
+        }
+        return ptr;
+    }
+
+    public static unsafe byte[] mstrdup(string str)
+    {
+        int len = System.Text.Encoding.UTF8.GetByteCount(str);
```

Comment:
Maybe in the future we can cache this at least for interned strings. https://learn.microsoft.com/en-us/dotnet/api/system.string.isinterned?view=net-8.0